packages/gamepads_android/android/src/main/kotlin/dev/markvideon/gamepads_android/GamepadsAndroidPlugin.kt

```diff
@@ -0,0 +1,95 @@
+package dev.markvideon.gamepads_android
+
+import androidx.annotation.NonNull
+import android.app.Activity
+import android.content.Context
+import android.hardware.input.InputManager
+import android.util.Log
+import android.view.InputDevice
+import android.view.KeyEvent
+import android.view.MotionEvent
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import kotlin.concurrent.thread
+
+class GamepadsAndroidPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
+  private lateinit var channel : MethodChannel
+  private lateinit var devices : DeviceListener
+  private lateinit var events : EventListener
+
+  private val TAG = "GamepadsAndroidPlugin"
+
+  private fun listGamepads(): List<Map<String, String>>  {
+    val results = mutableListOf<Map<String, String>>()
+    devices.getDevices().forEach({
+      results.add(mapOf(
+        "id" to it.key.toString(),
+        "name" to it.value.name
+      ))
+    })
+
+    return results
```

Comment:
this could be:
```
devices.getDevices().map { device ->
  mapOf {
      "id" to device.key.toString(),
      "name" to device.value.name,
  }
}
```